src/transformers/models/maskformer/modeling_maskformer.py

```diff
@@ -0,0 +1,2502 @@
+
+        Args:
+            in_features (`int`):
+                The number of input features (channels).
+            lateral_features (`int`):
+                The number of lateral features (channels).
+        """
+        super().__init__()
+        self.proj = nn.Sequential(
+            nn.Conv2d(lateral_features, in_features, kernel_size=1, padding=0, bias=False),
+            nn.GroupNorm(32, in_features),
+        )
+
+        self.block = MaskFormerFPNConvLayer(in_features, in_features)
+
+    def forward(self, down: Tensor, left: Tensor) -> Tensor:
+        left = self.proj(left)
+        down = interpolate(down, size=left.shape[-2:], mode="nearest")
+        down += left
+        down = self.block(down)
+        return down
+
+
+class MaskFormerFPNModel(nn.Module):
+    def __init__(self, in_features: int, lateral_widths: List[int], feature_size: int = 256):
+        """
+        Feature Pyramid Network, given an input tensor and a set of features map of different feature/spatial size, it
+        creates a list of features maps with the same feature size.
+
+        Args:
+            in_features (`int`):
+                The number of input features (channels).
+            lateral_widths (`List[int]`):
+                A list with the features (channels) size of each lateral connection.
+            feature_size (int, *optional*, defaults to `256`):
+                The features (channels) of the resulting feature maps.
+        """
+        super().__init__()
+        self.stem = MaskFormerFPNConvLayer(in_features, feature_size)
+        self.layers = nn.Sequential(
+            *[MaskFormerFPNLayer(feature_size, lateral_width) for lateral_width in lateral_widths[::-1]]
+        )
+
+    def forward(self, features: List[Tensor]) -> List[Tensor]:
+        fpn_features = []
+        last_feature = features[-1]
+        other_features = features[:-1]
+        output = self.stem(last_feature)
+        for layer, left in zip(self.layers, other_features[::-1]):
+            output = layer(output, left)
+            fpn_features.append(output)
+        return fpn_features
+
+
+class MaskFormerPixelDecoder(nn.Module):
+    def __init__(self, *args, feature_size: int = 256, mask_feature_size: int = 256, **kwargs):
+        """
+        Pixel Decoder Module proposed in [Per-Pixel Classification is Not All You Need for Semantic
+        Segmentation](https://arxiv.org/abs/2107.06278). It first runs the backbone's feature into a Feature Pyramid
+        Network creating a list of features maps. Then, it projects the last one to the correct `mask_size`.
+
+        Args:
+            feature_size (`int`, *optional*, defaults to `256`):
+                The feature size (channel dimension) of the FPN feature maps.
+            mask_feature_size (`int`, *optional*, defaults to `256`):
+                The features (channels) of the target masks size $C_{\epsilon}$ in the paper.
```

Comment:
Number should not be in code-blocks.

Also the inline LaTeX should not be between $ $ but \\\\( and \\\\)